pkg/autodiscovery/configresolver/configresolver.go

```diff
@@ -91,6 +93,30 @@ func Resolve(tpl integration.Config, svc listeners.Service) (integration.Config,
 	copy(resolvedConfig.InitConfig, tpl.InitConfig)
 	copy(resolvedConfig.Instances, tpl.Instances)
 
+	// Ignore the config from file if it's overridden by an empty config
+	// or by a different config for the same check
+	if tpl.Provider == names.File && svc.GetCheckNames() != "" {
+		checkNames := []string{}
+		err := json.Unmarshal([]byte(svc.GetCheckNames()), &checkNames)
+		if err != nil {
+			log.Debugf("Cannot parse check names: %v", err)
+		} else {
+			lenCheckNames := len(checkNames)
+			if lenCheckNames == 0 || (lenCheckNames == 1 && checkNames[0] == "") {
+				// Empty check names on k8s annotations or docker labels override the check config from file
+				// Used to deactivate unneeded OOTB autodiscovery checks defined in files
+				// The checkNames slice is considered empty also if it contains one single empty string
+				return resolvedConfig, fmt.Errorf("ignoring config from %s: another empty config is defined with the same AD identifier: %v", tpl.Source, tpl.ADIdentifiers)
+			}
+			for _, checkName := range checkNames {
+				if tpl.Name == checkName {
```

Comment:
since we're talking about 2 different services (== different containers), the first targeted `postgresql` container won't be labeled so the file config will be scheduled, the second one (labeled) will get the check from labels and override the file config.

Reply:
Are we sure we're OK with that behavior ?

Reply:
yes, for the specific use case that you mentioned, even before this PR we would have the same behavior, this PR only cares about multiple AD checks on 1 single service.

Reply:
@hkaj @olivielpeau 
based on the PR description and the comments, could you please confirm that we're OK with the overriding logic ?

Reply:
Thanks for the detailed PR description. I think the "new" behavior that you describe in the PR description makes sense and is better than the current behavior.


I have this additional question to be sure I completely understand the behavior you describe (both current and future behavior): when you refer to an `AD check from file`:
1. do you only refer to integration yamls that have an `ad_identifiers` key defined, or
2. do you also refer to any integration yaml defined in a file, including those that don't have an `ad_identifiers` key?

If the answer is `1.`, everything behaves as I'd expect :) .
If the answer to the question is `2.`, then like Maxime I think the behavior is unexpected, and I'd like to understand which use cases the behavior addresses.
In any case, I'm asking this question to better understand the general behavior, I don't think its answer affects the PR.

Reply:
thanks! the answer is `1.` as in AD we only try to resolve configs that have `ad_identifiers`

Reply:
Thanks for the extra info :+1: